services/venice-controller/src/main/java/com/linkedin/venice/controller/StoreBackupVersionCleanupService.java

```diff
@@ -104,6 +211,14 @@ protected boolean cleanupBackupVersion(Store store, String clusterName) {
     List<Version> versions = store.getVersions();
     List<Version> readyToBeRemovedVersions = new ArrayList<>();
     int currentVersion = store.getCurrentVersion();
+
+    // Do not delete version unless all routers and all servers are on same current version
+    if (multiClusterConfig.getControllerConfig(clusterName).isBackupVersionMetadataFetchBasedCleanupEnabled()
+        && (!validateAllRouterOnCurrentVersion(store, clusterName, currentVersion)
+            || !validateAllServerOnCurrentVersion(store, clusterName, currentVersion))) {
+      return false;
```

Comment:
Shall we emit a metric to track this behavior since it is unexpected?

Reply:
Sure, added a metric to track skipped version deletion.